RelationalAI/Rest.cs

```diff
@@ -41,15 +41,22 @@ public class Rest
 
         private readonly ILogger _logger;
 
-        public Rest(Context context, ILogger logger)
+        public Rest(Context context, ILogger logger, IAccessTokenHandler accessTokenHandler)
         {
             _context = context;
             HttpClient = new HttpClient();
             _logger = logger ?? new LoggerFactory().CreateLogger("RAI-SDK");
+
+            // Init AccessTokenHandler
+            AccessTokenHandler = accessTokenHandler ?? new DefaultAccessTokenHandler();
+            AccessTokenHandler.Rest = this;
+            AccessTokenHandler.Logger = _logger;
         }
 
         public HttpClient HttpClient { get; set; }
 
+        public IAccessTokenHandler AccessTokenHandler { get; set; }
```

Comment:
why is this public?

Reply:
the purpose is to make the Client class able to set the access token handler on the Rest class. Will change this field to make it internal
Otherwise we will need to use the Rest constructor the inject the AccessTokenHandler something like
```
 public Client(Context context, ILogger logger = null, IAccessTokenHandler accessTokenHandler = null)
        {
            _context = context;
            _logger = logger ?? new LoggerFactory().CreateLogger("RAI-SDK");
            _rest = new Rest(context, _logger, accessTokenHandler ?? new DefaultAccessTokenHandler(new Rest(context, null, null));
        }
```
vs (current implementation)

```
        public Client(Context context, ILogger logger = null, IAccessTokenHandler accessTokenHandler = null)
        {
            _context = context;
            _logger = logger ?? new LoggerFactory().CreateLogger("RAI-SDK");
            _rest = new Rest(context, _logger);
            _rest.AccessTokenHandler = accessTokenHandler ?? new DefaultAccessTokenHandler(_rest);
        }
```

I lean toward using the second option